[Scout] Align skills and docs with current Scout behavior (#281056)
elastic/kibana: .agents/skills/scout-ui-testing/SKILL.md

    .waitFor({ state: 'visible' });

  // ❌ page object — do not use expect
  await includeEmptyRows.click();
  await expect(includeEmptyRows).toHaveAttribute('aria-checked', 'true');
  ```
  This applies to any attribute check (`aria-pressed`, `aria-selected`, `aria-expanded`, `disabled`, etc.). The `.and()` locator composition auto-retries until the combined selector matches, equivalent to an assertion but without pulling `expect` into the page object.
- **Keep route mocks out of page objects** — page objects are for UI interactions only. Put `page.route()` mocks in a dedicated `fixtures/mocks.ts` file as standalone functions that accept `page` as a parameter. See `cloud_security_posture/test/scout_cspm_agentless/ui/fixtures/mocks.ts` for the reference pattern.
- Don't make API calls from page objects (use `apiServices`/`kbnClient` in hooks instead).
- Register plugin page objects by extending the `pageObjects` fixture in `test/scout*/ui/fixtures/index.ts`.
- **Use `readonly` class fields for static locators** — assign them in the constructor, not as getter methods. Use methods only for parameterized locators/actions. See `DashboardApp` in `kbn-scout` for the reference pattern.

> - **EUI components — prefer published Component Objects over raw selectors and legacy wrappers.** Drive EUI widgets through `page.components.*` (e.g. `page.components.comboBox(testSubj)`) — Scout's factories over `@elastic/eui-test-helpers`. Don't 1:1-map an old wrapper API: use only the interactions the test needs and push data-correctness checks to API/unit tests.

Reviewer: ```suggestion
- **EUI components — prefer published EUI Test Helpers over raw selectors and legacy wrappers.** Drive EUI widgets through `page.components.*` (e.g. `page.components.comboBox(testSubj)`) — Scout's factories over `@elastic/eui-test-helpers`. Don't 1:1-map an old wrapper API: use only the interactions the test needs and push data-correctness checks to API/unit tests.
```

> - **Compatibility fallback only.** When no equivalent Component Object exists, an existing `Eui*Wrapper` class from `@kbn/scout` may be used temporarily. Do not add or extend wrappers in a test suite. Route missing Component Object capabilities through the shared Apps DX/EUI contribution workflow.

Reviewer: ```suggestion
- **Compatibility fallback only.** When no equivalent EUI test helper exists, fallback to using locators. Do not add or extend wrappers in a test suite. Route missing Component Object capabilities through the shared Apps DX/EUI contribution workflow.
```

Scout EUI wrappers are deprecated, see https://github.com/elastic/kibana/pull/284005
